src/Interactive/Host/Interactive/Core/InteractiveHost.cs

```diff
@@ -413,25 +369,38 @@ private async Task<InitializedRemoteService> TryGetOrCreateRemoteServiceAsync()
             return default;
         }
 
-        private async Task<TResult> Async<TResult>(Action<Service, RemoteAsyncOperation<TResult>> action)
+        private async Task<TResult> Async<TResult>(string targetName, params object[] arguments)
         {
+            var initializedService = await TryGetOrCreateRemoteServiceAsync().ConfigureAwait(false);
+            TaskCompletionSource<TResult> _completion = new TaskCompletionSource<TResult>();
+            EventHandler _processExitedHandler = new EventHandler((_, __) =>
+            {
+                initializedService.ServiceOpt.Process.Exited -= _processExitedHandler;
+                _completion.TrySetResult(default(TResult));
+            });
```

Comment:
Just realized we don't need this handler. The reason why it was used before was to detect the case when the process is terminated before it completes the remote operation.
JSON-RPC should be able to handle that.